### PR TITLE
Mark num_classes as required in Classifier classes

### LINKED ISSUE #955: Require `num_classes` to be set when creating a classifier
When creating a classifier via `from_preset()` or the default constructor, we should require `num_classes` to be passed. This will be consistent with recent updates to keras-cv.

```python
model = BertClassifier.from_preset("bert_base_en_uncased")   # Error, no num_classes.
model = BertClassifier(backbone)  # Error, no num_classes.
```

### PR DESCRIPTION
**Pull request by chenmoneygithub**
resolve #955

### CODE REVIEW

**Comment by mattdangerw** [comment]
There are some legit failures in our `"large"` testing, take a look!

File: keras_nlp/models/distil_bert/distil_bert_classifier.py [hunk 1]
```diff
@@ -210,3 +210,18 @@ def preprocessor_cls(cls):
     @classproperty
     def presets(cls):
         return copy.deepcopy(backbone_presets)
+
+    @classmethod
+    def from_preset(
+        cls,
+        preset,
+        num_classes,
```
Comment (mattdangerw) [inline keras_nlp/models/distil_bert/distil_bert_classifier.py:218]: I think if we just follow @jbischof's work in CV here, we should not extend `from_preset` everywhere, and just rely on the fact we will get a somewhat descriptive error if `num_classes` is not provider via kwargs.

There is probably some work we could do with a try/catch block on the base `Task` class to improve the error message if we would like, but I don't think we want to redefine `from_preset` everywhere. The goal should be keeping the end task classes, as short and concise as we can!

**Review by mattdangerw — CHANGES_REQUESTED** [review]
Looks good overall! One major comment and some test failures.

File: keras_nlp/models/albert/albert_presets_test.py [hunk 2]
```diff
@@ -115,7 +117,13 @@ def test_preset_docstring(self, cls):
     def test_unknown_preset_error(self, cls):
         # Not a preset name
         with self.assertRaises(ValueError):
-            cls.from_preset("albert_base_en_uncased_clowntown")
+            if cls == AlbertClassifier:
+                cls.from_preset(
+                    "albert_base_en_uncased_clowntown",
+                    num_classes=2,
```
[the remark this answers is not in the document]
Comment (chenmoneygithub) [reply keras_nlp/models/albert/albert_presets_test.py]: good idea, done

**Pull request merged**